src/parser/semantic_check.c

```diff
@@ -11457,6 +11464,13 @@ pt_assignment_compatible (PARSER_CONTEXT * parser, PT_NODE * lhs, PT_NODE * rhs)
 			  else
 			    {
 			      d = tp_domain_resolve_default (lhs_dbtype);
+			      if (lhs->type_enum == PT_TYPE_NUMERIC && lhs->data_type != NULL)
```

Comment:
checking lhs->data_type != NULL is not needed. lhs->data_type is not accessed in the if-block.